_plugins/code_block_validator.rb

```diff
@@ -0,0 +1,35 @@
+# Description : This plugin checks code blocks created using `or``` 
+# in their markdown state for special characters that can cause problem in terminal.
+# Authour : Reza R <54559947+frozenprocess@users.noreply.github.com>
+#####
+Jekyll::Hooks.register :pages, :pre_render do |post|
+    # do nothing if url indicates legal folder
+    if !post.path.match(/\/legal\//)
```

Comment:
This is a pretty large branch, does ruby have return statements?

Reply:
Ruby has returns but unfortunately it is not possible to use it in Jekyll hooks, such instruction triggers an exception.
I'll see what else can be used here.

Reply:
The only thing that I could come up with was getting rid of line 7 `if !post.path.match(/\/legal\//)` and adding it to line 10 `post.content.scan(/(?:[`]{3}(.*?)[`]{3})/m)` transforming it to an if statement with `&&` operator to first evaluate the path and then check for regex pattern.
@joshti WTDY?

Reply:
Nah this makes more-sense this way.